### PR TITLE
SW-5625 Database schema for applications

### PR DESCRIPTION
**Pull request by sgrimm**
Define the database representation of accelerator applications.

### CODE REVIEW

**Comment by sgrimm** [comment]
* **#2143** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2143?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2142** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2142?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2150** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2150?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2139** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2139?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#2149** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2149?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @sgrimm and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: src/main/resources/db/migration/0250/V286__Applications.sql [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+CREATE TABLE accelerator.applications (
+    id BIGINT PRIMARY KEY GENERATED BY DEFAULT AS IDENTITY,
+    project_id BIGINT NOT NULL REFERENCES projects ON DELETE CASCADE,
+    planting_site_id BIGINT NOT NULL REFERENCES tracking.planting_sites,
+    internal_name TEXT NOT NULL,
+    created_by BIGINT NOT NULL REFERENCES users,
+    created_time TIMESTAMP WITH TIME ZONE NOT NULL,
+    modified_by BIGINT NOT NULL REFERENCES users,
+    modified_time TIMESTAMP WITH TIME ZONE NOT NULL,
+    status_id INTEGER NOT NULL REFERENCES accelerator.application_statuses,
```
Comment (tommylau523) [inline src/main/resources/db/migration/0250/V286__Applications.sql]: Do we need the status as a column, or should we just look for the latest in history?

File: src/main/resources/db/migration/0250/V286__Applications.sql [hunk 2]
```diff
@@ -0,0 +1,48 @@
+CREATE TABLE accelerator.application_statuses (
+    id INTEGER PRIMARY KEY,
+    name TEXT NOT NULL UNIQUE
+);
+
+CREATE TABLE accelerator.applications (
+    id BIGINT PRIMARY KEY GENERATED BY DEFAULT AS IDENTITY,
+    project_id BIGINT NOT NULL REFERENCES projects ON DELETE CASCADE,
```
Comment (tommylau523) [inline src/main/resources/db/migration/0250/V286__Applications.sql]: We can add unique to project_id to make sure 1:1 application to project ratio

File: src/main/resources/db/migration/0250/V286__Applications.sql (hunk 1 above)
Comment (sgrimm) [reply src/main/resources/db/migration/0250/V286__Applications.sql]: Having it in the table will make it easier to expose applications via the search API (no need for a view or a subquery). More efficient too, though I expect the frequency of use and the data volume will be small enough that that's not too big a factor in practice.

File: src/main/resources/db/migration/0250/V286__Applications.sql (hunk 2 above)
Comment (sgrimm) [reply src/main/resources/db/migration/0250/V286__Applications.sql]: Good call, yeah.

File: src/main/resources/db/migration/0250/V286__Applications.sql (hunk 1 above)
Comment (tommylau523) [reply src/main/resources/db/migration/0250/V286__Applications.sql]: Either way! I was just thinking about keeping data independent in theory but this is easier to query for sure!

**Comment by sgrimm** [comment]
I made a couple tweaks and I think this is ready for review as an initial version.

**Review by tommylau523 — APPROVED** [review]

**Comment by sgrimm** [comment]
### Merge activity

* **Jul 12, 1:34 PM EDT**: @sgrimm started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-server/2139).
* **Jul 12, 1:36 PM EDT**: [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-server/2139) rebased this pull request as part of a merge.
* **Jul 12, 1:44 PM EDT**: @sgrimm merged this pull request with [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-server/2139).

**Pull request merged**